biogeochem/EDPhysiologyMod.F90

```diff
@@ -1866,22 +1879,30 @@ subroutine SeedIn( currentSite, bc_in )
                 end select
 
                 ! Seed input from external sources (user param seed rain, or dispersal model)
-                seed_in_external =  seed_stoich*EDPftvarcon_inst%seed_suppl(pft)*years_per_day
+                ! Include both prescribed seed_suppl and seed_in dispersed from neighbouring gridcells
+                seed_in_external = bc_in%seed_in(pft)/area + seed_stoich*EDPftvarcon_inst%seed_suppl(pft)*years_per_day ![kg/m2/day]
                 litt%seed_in_extern(pft) = litt%seed_in_extern(pft) + seed_in_external
 
                 ! Seeds entering externally [kg/site/day]
-                site_mass%seed_in = site_mass%seed_in + seed_in_external*currentPatch%area
+                site_mass%seed_in = site_mass%seed_in + seed_in_external*currentPatch%area ![kg/site/day]
              end if !use this pft
           enddo
-
-
+          
           currentPatch => currentPatch%younger
        enddo
 
+        do pft = 1,numpft
+            site_mass%seed_out = site_mass%seed_out + site_seed_rain(pft)*site_disp_frac(pft) ![kg/site/day]
+        end do
+ 
+    end do
+
+    do pft = 1,numpft
+        bc_out%seed_out(pft) = bc_out%seed_out(pft) + site_seed_rain(pft)*site_disp_frac(pft) ![kg/site/day]
```

Comment:
@YanlanLiu I think this also should be brought inside the element loop since `site_seed_rain` is re-calculated for each element.  Right now the code as written will use the last `site_seed_rain` calculation for the final element, not all the elements.

Reply:
@ckoven agreed here as well.